godot-core/src/meta/args/as_arg.rs

```diff
@@ -764,3 +764,37 @@ where
 
 #[doc(hidden)] // Easier for internal use.
 pub type ToArg<'r, Via, Pass> = <Pass as ArgPassing>::Output<'r, Via>;
+
+/// This type exists only as a place to add doctests for `AsArg`, which do not need to be in the public documentation.
+///
+/// `AsArg<Option<Gd<UserClass>>` can be used with signals correctly:
+///
+/// ```no_run
+/// # use godot::prelude::*;
+/// #[derive(GodotClass)]
+/// #[class(init, base = Node)]
+/// struct MyClass {
+///     base: Base<Node>
+/// }
+///
+/// #[godot_api]
+/// impl MyClass {
+///     #[signal]
+///     fn signal_optional_user_obj(arg1: Option<Gd<MyClass>>);
+///
+///     fn foo(&mut self) {
+///         let arg = self.to_gd();
+///         // Directly:
+///         self.signals().signal_optional_user_obj().emit(&arg);
+///         // Via Some:
+///         self.signals().signal_optional_user_obj().emit(Some(&arg));
+///         // With move:
+///         self.signals().signal_optional_user_obj().emit(Some(arg).as_ref());
+///         // With None (Note: Gd::null_arg() is restricted):
+///         self.signals().signal_optional_user_obj().emit(None::<Gd<MyClass>>.as_ref());
```

Comment:
Also here?

What do you mean with "restricted", could you clarify?

Reply:
Limited to engine classes